conda/reporters.py

```diff
@@ -6,27 +6,56 @@
 
 from __future__ import annotations
 
+import logging
+import sys
+from functools import lru_cache
+from typing import TYPE_CHECKING
+
+from .base.constants import (
+    DEFAULT_JSON_REPORTER_BACKEND,
+)
 from .base.context import context
 
+if TYPE_CHECKING:
+    from typing import Callable
 
-def render(data, style: str | None = None, **kwargs) -> None:
-    for settings in context.reporters:
-        reporter = context.plugin_manager.get_reporter_backend(settings.get("backend"))
-        output = context.plugin_manager.get_reporter_output(settings.get("output"))
+logger = logging.getLogger(__name__)
+
+
+@lru_cache
```

Comment:
Since the default size of the lru_cache may be larger depending on the objects handled by the function

```suggestion
@lru_cache(maxsize=None)
```